GoogleForCreators/web-stories-wp
# Design System: Popup clean up

## Context
This is some follow up work related to the Layers Panel `Tooltip` getting negative valued x-offsets. Since the Layers Panel has some very nested absolute positioning the `getOffset` util thinks the `Tooltip` should be based off somewhere that the panel isn't. `Tooltip` offset updates, edge cases, etc where fixed in #11280. However, we still needed to address the `getOffset` to account for edge cases in `Popups` that were a work-around for those Tooltips.    

We had a special prop flag `resetXOffset`, to know when to look for the edge cases where this was an issue. Now that `Tooltip` is separate we can remove the `resetXOffset` flag and better handle any off screen issues directly in the `Popup` component. 


## Summary


`Popup` now updates the x-offset in cases where it would either collide with admin side nav, or otherwise get cutoff screen in x. 

## Relevant Technical Choices
Removed `resetXOffset` from the shared util `getOffset`. Both `Tooltip` and `Popup` use this function to grab offset values for portals. I wanted to keep this portion straightforward.  Since `Tooltip` and `Popup` have varied `placement` considerations I wanted to keep any special offset updates or edge case logic specific to each component. The `getOffset` will still give us `maxOffsetX` to prevent us from cutoff on the right edge, however,  `Popup` now takes `leftOffset` into consideration for RTL mode. `leftOffset` is also being utilized for LTR sidenav bar consideration. 

~~The bulk of these files was just adding `isRTL` and `leftOffset` to all uses of `Popup`.~~

New `usePopup` hook and `PopupProvider`. This provider now gives us access to `leftOffset`, `topOffset` and `isRTL` so we no longer need to pass these as props to `Tooltip` or `Popup` 🎉
 


## To-do



## User-facing changes


Now that `Insertion Menu` is receiving `isRTL` the popup placement is correct in the RTL view as seen below. 
|Before|After|
|--|--|
|<img width="404" alt="Screen Shot 2022-04-21 at 8 31 11 AM" src="https://user-images.githubusercontent.com/1820266/164481024-6c88945a-3ad9-4aac-9c82-a29532139ae3.png">|<img width="403" alt="Screen Shot 2022-04-21 at 8 29 54 AM" src="https://user-images.githubusercontent.com/1820266/164480930-90a957e6-ad47-4db4-a757-693d81da93f2.png">|


## Testing Instructions






This PR can be tested by following these steps:

1. In RTL, open the Insert panel and hover over an image. 
2. Click the `+` icon  to open the `Insertion Menu`
3. The Popup should no longer be cutoff by the wp-admin bar
4. While we are here, let/s check for Popup regressions. 
5. Select / add text on the canvas and open the `Style` Panel
6. Open the color picker, (any version of it),  save styles, Font dropdown
7. We can also check the dropdowns in Document tab and the Publish Time Popup. 
8. Double check Dropdowns in the Dashboard and Dashboard settings as well.  


## Reviews

### Does this PR have a security-related impact?
no


### Does this PR change what data or activity we track or use?
no


### Does this PR have a legal-related impact?
no


## Checklist




---



Fixes #11201 
Fixes #11303
